server/controllers/users.controller.js

```diff
@@ -236,20 +236,50 @@ const getCurrentUserDataFromToken = async (req, res, _) => {
 
 const updateUserData = async (req, res, _) => {
   try {
-    const { value, error } = userRegisterReqBodySchema.validate(req.body);
-    const { email, password, firstName } = value;
+    const { value, error } = userUpdateReqBodySchema.validate(req.body);
+    const { email, password, firstName, userCurrency } = value;
 
     if (error) {
       return res.status(400).json({ status: 'error', code: 400, message: error.message });
     }
 
     const id = req.user.id;
     const capitalizedFirstName = capitalizeEachWord(firstName);
-    await updateUserDataByIdInDB(id, { email, password, firstName: capitalizedFirstName });
+
+    if (password === 'samePass') {
+      await updateUserDataByIdInDB(id, {
+        firstName: capitalizedFirstName,
+        userCurrency,
+      });
+
+      return res.json({
+        status: 'success',
+        code: 200,
+        message: `User's data successfully updated.`,
+        user: {
+          email,
+          firstName: capitalizedFirstName,
+          userCurrency,
+        },
+      });
+    }
+
+    const hashedPassword = await hashPassword(password);
```

Comment:
Co tutaj zachodzi? Wygląda mi na to, że dostajemy zamiast hasła komunikat samePass, a później go hashujemy?

Reply:
Jak wejdziesz do updateUserModal, to tam jest, kod, którydziała mniej-więcej tak:
w przypadku, jeżeli hasło nie zostaje zmienione (czyli user zostawia pusty string w inpucie password), to "puszczam" w tym wypadku przez dispatch updateUser password: "samePass"- ze względy na to, że funkcja backendowa wymaga (w joi) określonego schematu - hasła: 6-12 znaków. Następnie daję warunek w if - jeżeli password === 'samePass' -zostawiam hasło w spokoju, przesyłając tylko firstName i userCurrency do bazy :)

Reply:
Ok, rozumiem :)